cmd/api.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"context"
+
+	httpAPI "github.com/SigNoz/signoz-otel-collector/pkg/api/http"
+	"github.com/SigNoz/signoz-otel-collector/pkg/log"
+	"github.com/SigNoz/signoz-otel-collector/pkg/server/http"
+	"github.com/SigNoz/signoz-otel-collector/pkg/storage"
+	"github.com/spf13/cobra"
+)
+
+func registerApi(app *cobra.Command) {
+	var adminHttpConfig adminHttpConfig
+	var storageConfig storageConfig
+
+	cmd := &cobra.Command{
+		Use:   "api",
+		Short: "Starts an api to interact with configuration",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Create the logger by taking the log-level from the root command
+			logger := log.NewZapLogger(app.PersistentFlags().Lookup("log-level").Value.String())
+			defer func() {
+				_ = logger.Flush()
+			}()
+
+			return runApi(
+				cmd.Context(),
+				logger,
+				storageConfig,
+				adminHttpConfig,
+			)
+		},
+	}
+
+	adminHttpConfig.registerFlags(cmd)
+	storageConfig.registerFlags(cmd)
```

Comment:
What do you think about adding a config file instead for API options? That would remain consistent with the rest of the collector components and make adding more options for sub-components separate from the high-level cli flags.

Reply:
I was initially thinking of the same but managing credentials is a pain. The precedence defined currently is:
- flag
- env

We can add a config file precedence to this but I still would like to keep the flags and env.
- flag
- env
- config file

Reply:
Why is managing credentials pain?

Reply:
How do you inject credentials into a container using any orchestration engine?

Reply:
The config file will reference an env like `password: ${PASSWORD}` and the env gets injected before the config gets applied?